mbed_tools_ci/utils/filesystem_helpers.py

```diff
@@ -77,3 +80,52 @@ def find_file_in_tree(file_name: str,
             f'File [{file_name}] not found anywhere in directories {"above" if top else "under"} {starting_point}'
         )
     return file_path
+
+
+class TemporaryDirectory:

```

Comment:
Have you seen https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory ? It might cover our usecase here.

Reply:
After our conversation it seems `TemporaryDirectory` doesn't work on windows with certain files, it fails on cleanup. Which is annoying. There's an issue on the python bug-tracker from 2016: https://bugs.python.org/issue26660

Reply:
I referenced this issue in the doc string so that we remember the rationale